src/vdxf/classes/requestobjects/AppEncryptionRequestDetails.ts

```diff
@@ -0,0 +1,251 @@
+  }
+
+  hasFromAddress(): boolean {
+    return this.flags.and(AppEncryptionRequestDetails.HAS_FROM_ADDRESS).gt(new BN(0));
+  }
+
+  hasToAddress(): boolean {
+    return this.flags.and(AppEncryptionRequestDetails.HAS_TO_ADDRESS).gt(new BN(0));
+  }
+
+  getByteLength(): number {
+    let length = 0;
+
+    length += varint.encodingLength(this.flags);
+
+    // encryptToKey - zaddress encoding (43 bytes for sapling address data)
+    length += 43; // Sapling address decoded data (11 + 32 bytes)
+
+    length += varint.encodingLength(this.derivationNumber);
+
+    if (this.hasSecondarySeedDerivation()) {
+      length += varint.encodingLength(this.secondaryDerivationNumber);
+    }
+
+    if (this.hasFromAddress()) {
+        length += this.fromAddress.getByteLength();
+    }
+
+    if (this.hasToAddress()) {
+       length += this.toAddress.getByteLength();
+    }
+
+    return length;
+  }
+
+  toBuffer(): Buffer {
+    // Set flags before serialization
+    this.setFlags();
+
+    const writer = new BufferWriter(Buffer.alloc(this.getByteLength()));
+
+    // Write flags
+    writer.writeVarInt(this.flags);
+
+    // Write encryptToAddress as decoded sapling address data
+    const saplingData = decodeSaplingAddress(this.encryptToZAddress);
+    writer.writeSlice(Buffer.concat([saplingData.d, saplingData.pk_d]));
+
+    // Write mandatory derivation number
+    writer.writeVarInt(this.derivationNumber);
+
+    if (this.hasSecondarySeedDerivation()) {
+      writer.writeVarInt(this.secondaryDerivationNumber);
+    }
+
+    if (this.hasFromAddress()) {
+        writer.writeSlice(this.fromAddress.toBuffer());
+    }
+
+    if (this.hasToAddress()) {
+       writer.writeSlice(this.toAddress.toBuffer());
+    }
+
+    return writer.buffer;
+  }
+
+  fromBuffer(buffer: Buffer, offset?: number): number {
+    const reader = new BufferReader(buffer, offset);
+
+    // Read flags
+    this.flags = reader.readVarInt();
+
+    // Read encryptToAddress as 43-byte sapling data and encode as sapling address
+    const saplingData = reader.readSlice(43);
+    this.encryptToZAddress = toBech32('zs', saplingData);
+
+    // Read mandatory derivation number
+    this.derivationNumber = reader.readVarInt();
+
+    // Read optional derivation number if flag is set
+    if (this.hasSecondarySeedDerivation()) {
+      this.secondaryDerivationNumber = reader.readVarInt();
+    }
+
+    // Read addresses based on flags
+    if (this.hasFromAddress()) {
+      const CompactId = new CompactIdAddressObject();
+      reader.offset = CompactId.fromBuffer(reader.buffer, reader.offset);
+      this.fromAddress = CompactId;
+    }
+
+    if (this.hasToAddress()) {
+      const CompactId = new CompactIdAddressObject();
+      reader.offset = CompactId.fromBuffer(reader.buffer, reader.offset);
+      this.toAddress = CompactId;
+    }
+
+    return reader.offset;
+  }
+
+  toJSON(): AppEncryptionRequestDetailsJson {
```

Comment:
from/to json are written as fromJson and toJson elsewhere, so we should keep it consistent. This will also cause an error when added as an ordinalvdxfobject type because it expects toJson